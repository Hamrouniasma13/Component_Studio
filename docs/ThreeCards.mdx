---
name: Three cards (deprecated)
route: /three-cards
menu: 🗑️ Deprecated
---

import { Playground } from 'docz'
import { ThreeCards } from '../src'

# Three cards

## Colors

- `#f5f5f5` for background color
- `#fff` for image border
- `theme.colors.text` for title and text
- `theme.colors.primary` for button background

## Example

<Playground fullBleed noPadding>
  <ThreeCards
    title="My Cards"
    cards={[
      {
        image: {
          src:
            'http://eb-employee-fr.cdn.prismic.io/eb-employee-fr%2Fcd8d94f6-734c-404f-ad4a-ae88117144ff_e.svg',
          alt: 'My Easy Sante',
        },
        title: 'My Easy Sante',
        text:
          'Pour préserver et améliorer sa santé au quotidien avec My Easy Santé. C’est utile et efficace !',
        action: {
          url: 'https://fr.wikipedia.org/wiki/Pyramides_de_Gizeh',
          label: 'Découvrir',
        },
      },
      {
        image: {
          src:
            'http://eb-employee-fr.cdn.prismic.io/eb-employee-fr%2Fc50bf496-be89-46d3-ad0d-f72f4b5679fd_f.svg',
          alt: 'Appeler un médecin 24/7',
        },
        title: 'Appeler un médecin 24/7',
        text:
          "3633 pour consulter immédiatement un médecin par téléphone (prix d'un appel local) ou par vidéo (+33 1 55 92 27 54 depuis l'étranger).",
      },
      {
        image: {
          src:
            'http://eb-employee-fr.cdn.prismic.io/eb-employee-fr%2F445e7e8d-89cd-4d51-8c1e-cac8ae3d07e2_g.svg',
          alt: 'Accéder au réseau de 6000 professionnels Santé Itelis',
        },
        title: 'Accéder au réseau de 6000 professionnels Santé Itelis',
        text:
          '6 000 professionnels de la santé Itelis conseillent et facilitent l’accès aux soins à plus de 10 millions de personnes.(opticiens, dentistes, audioprothésistes...)',
        action: {
          url: 'http://www.chateau-amboise.com/',
          label: 'Découvrir',
        },
      },
    ]}
  />
</Playground>

<Playground fullBleed noPadding>
  <ThreeCards
    title="Features"
    cards={[
      {
        image: {
          src: 'https://wanchain.org/dist/f1cc29e1c8dbe68a58cecc19cee10b66.png',
          alt: 'Easier',
        },
        title: 'Easier',
        text: 'Cupidatat ad adipisicing occaecat eiusmod.',
        action: {
          url: 'https://fr.wikipedia.org/wiki/Pyramides_de_Gizeh',
          label: 'Découvrir',
        },
      },
      {
        image: {
          src: 'https://wanchain.org/dist/c66ea9fab59af9487eee15564c9f7d6a.png',
          alt: 'Faster',
        },
        title: 'Faster',
        text:
          'Velit elit ullamco minim culpa sunt et labore anim irure voluptate in aliqua qui aliquip',
      },
      {
        image: {
          src: 'https://wanchain.org/dist/2fe4aeea1b3a12c3a860121a71f52b3a.png',
          alt: 'Stronger',
        },
        title: 'Stronger',
        text:
          'Do nostrud est irure sunt amet dolor consequat velit elit pariatur adipisicing est.',
        action: {
          url: 'http://www.chateau-amboise.com/',
          label: 'Découvrir',
        },
      },
    ]}
  />
</Playground>
