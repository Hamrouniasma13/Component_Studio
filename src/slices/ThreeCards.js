/** @jsx jsx */
import { jsx, ThemeContext } from '@emotion/core'
import React from 'react'
import PropTypes from 'prop-types'
import SlickSlider from 'react-slick'
import ProgressiveImage from 'react-progressive-image'

import Slice from './Slice'
import Text from '../components/Text'
import Button from '../components/Button'

function Card({ image, title, text, action }) {
  const theme = React.useContext(ThemeContext)

  const styles = {
    wrapper: {
      backgroundColor: '#fff',
      border: `1px solid ${theme.colors.lightGray}`,
      height: '600px',
      overflow: 'hidden',
      width: '320px',
      margin: '0 auto',
      padding: '30px',
      textAlign: 'center',
      display: 'flex',
      color: theme.colors.text,
      flexDirection: 'column',

      [theme.media.tablet]: {
        width: '333px', // let space for boxShadow
        marginLeft: '10px'
      },

      [theme.media.laptop]: {
        width: '350px',
        marginLeft: 'auto'
      }
    },
    image: {
      willChange: 'opacity',
      transition: 'opacity 0.5s',
      maxHeight: '200px',
      minHeight: '140px',
      margin: '20px auto'
    }
  }

  return (
    <div css={styles.wrapper}>
      <div css={{ marginBottom: 30 }}>
        <ProgressiveImage src={image.src}>
          {(src, loading) => (
            <img
              css={{ ...styles.image, opacity: loading ? 0 : 1 }}
              src={src}
              alt={image.alt}
            />
          )}
        </ProgressiveImage>
      </div>
      <Text.h3 mb={30}>{title}</Text.h3>
      <Text css={{ textAlign: 'center', flex: 1 }}>{text}</Text>
      {action ? (
        <div>
          <Button href={action.url} color="secondary">
            {action.label}
          </Button>
        </div>
      ) : null}
    </div>
  )
}

Card.propTypes = {
  image: PropTypes.shape({
    src: PropTypes.string.isRequired,
    alt: PropTypes.string
  }),
  title: PropTypes.string,
  text: PropTypes.any,
  action: PropTypes.shape({
    label: PropTypes.string,
    url: PropTypes.string
  })
}

export default function ThreeCards({ title, cards }) {
  const theme = React.useContext(ThemeContext)

  const styles = {
    wrapper: {
      marginLeft: '-15px',
      marginRight: '-15px',
      paddingBottom: '30px', // Space for dots
      [theme.media.tablet]: {
        marginLeft: 0,
        marginRight: 0
      }
    }
  }

  const sliderSettings = {
    dots: true,
    arrows: false,
    slidesToShow: 3,
    responsive: [
      {
        breakpoint: theme.breakpoints.desktop - 1,
        settings: { slidesToShow: 2 }
      },
      {
        breakpoint: theme.breakpoints.tablet - 1,
        settings: { slidesToShow: 1 }
      }
    ]
  }

  return (
    <Slice title={title}>
      <div css={styles.wrapper}>
        <SlickSlider {...sliderSettings}>
          {cards && cards.map((card, key) => <Card {...card} key={key} />)}
        </SlickSlider>
      </div>
    </Slice>
  )
}

ThreeCards.propTypes = {
  title: Slice.propTypes.title,
  cards: PropTypes.arrayOf(PropTypes.shape(Card.propTypes)).isRequired
}
