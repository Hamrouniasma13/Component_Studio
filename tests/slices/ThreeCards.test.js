import React from 'react';
import renderer from 'react-test-renderer';

import ThreeCards from '../../slices/ThreeCards';

describe('Component: <ThreeCards />', () => {
  it('render snapshot', () => {
    const tree = renderer
      .create(
        <ThreeCards
          title="My Cards"
          cards={[
            {
              image: {
                src:
                  'http://eb-employee-fr.cdn.prismic.io/eb-employee-fr%2Fcd8d94f6-734c-404f-ad4a-ae88117144ff_e.svg',
                alt: 'My Easy Sante'
              },
              title: 'My Easy Sante',
              text:
                'Pour préserver et améliorer sa santé au quotidien avec My Easy Santé. C’est utile et efficace !',
              action: {
                url: 'https://fr.wikipedia.org/wiki/Pyramides_de_Gizeh',
                label: 'Découvrir'
              }
            },
            {
              image: {
                src:
                  'http://eb-employee-fr.cdn.prismic.io/eb-employee-fr%2Fc50bf496-be89-46d3-ad0d-f72f4b5679fd_f.svg',
                alt: 'Appeler un médecin 24/7'
              },
              title: 'Appeler un médecin 24/7',
              text:
                "3633 pour consulter immédiatement un médecin par téléphone (prix d'un appel local) ou par vidéo (+33 1 55 92 27 54 depuis l'étranger)."
            },
            {
              image: {
                src:
                  'http://eb-employee-fr.cdn.prismic.io/eb-employee-fr%2F445e7e8d-89cd-4d51-8c1e-cac8ae3d07e2_g.svg',
                alt: 'Accéder au réseau de 6000 professionnels Santé Itelis'
              },
              title: 'Accéder au réseau de 6000 professionnels Santé Itelis',
              text:
                '6 000 professionnels de la santé Itelis conseillent et facilitent l’accès aux soins à plus de 10 millions de personnes.(opticiens, dentistes, audioprothésistes...)',
              action: {
                url: 'http://www.chateau-amboise.com/',
                label: 'Découvrir'
              }
            }
          ]}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // it('render simpliest ThreeCards', () => {
  //   const tree = renderer
  //     .create(
  //       <ThreeCards
  //         cards={[
  //           {
  //             image: {
  //               url: 'https://picsum.photos/546/970?image=670',
  //               alt: 'Achitechture',
  //               dimensions: { width: 238, height: 380 }
  //             },
  //             title: 'Médecine et soins courants'
  //           },
  //           {
  //             image: {
  //               url: 'https://picsum.photos/546/970?image=903',
  //               alt: 'Achitechture',
  //               dimensions: { width: 238, height: 380 },
  //               mobile: {
  //                 alt: 'architecture',
  //                 url: 'https://picsum.photos/546/307?image=903',
  //                 dimensions: { width: 546, height: 307 }
  //               },
  //               tablet: {
  //                 alt: 'architecture',
  //                 url: 'https://picsum.photos/737/415?image=903',
  //                 dimensions: { width: 737, height: 415 }
  //               },
  //               desktop: {
  //                 alt: 'architecture',
  //                 url: 'https://picsum.photos/238/380?image=903',
  //                 dimensions: { width: 238, height: 380 }
  //               }
  //             },
  //             links: [
  //               {
  //                 url: 'https://fr.wikipedia.org/wiki/Pyramides_de_Gizeh',
  //                 label: 'En savoir plus'
  //               },
  //               {
  //                 url: 'https://fr.wikipedia.org/wiki/Pyramides_de_Gizeh',
  //                 label: 'Estimer mon reste à charge'
  //               }
  //             ]
  //           },
  //           {
  //             image: {
  //               url: 'https://picsum.photos/546/970?image=718',
  //               alt: 'Achitechture',
  //               dimensions: { width: 238, height: 380 },
  //               mobile: {
  //                 alt: 'architecture',
  //                 url: 'https://picsum.photos/546/307?image=718',
  //                 dimensions: { width: 546, height: 307 }
  //               },
  //               tablet: {
  //                 alt: 'architecture',
  //                 url: 'https://picsum.photos/737/415?image=718',
  //                 dimensions: { width: 737, height: 415 }
  //               },
  //               desktop: {
  //                 alt: 'architecture',
  //                 url: 'https://picsum.photos/238/380?image=718',
  //                 dimensions: { width: 238, height: 380 }
  //               }
  //             },
  //             title: 'Optique',
  //             text:
  //               'Prise en charge à partir de #100€ pour les montures et #160€ pour les verres.'
  //           },
  //           {
  //             image: {
  //               url: 'https://picsum.photos/546/970?image=433',
  //               alt: 'Achitechture',
  //               dimensions: { width: 238, height: 380 },
  //               mobile: {
  //                 alt: 'architecture',
  //                 url: 'https://picsum.photos/546/307?image=433',
  //                 dimensions: { width: 546, height: 307 }
  //               },
  //               tablet: {
  //                 alt: 'architecture',
  //                 url: 'https://picsum.photos/737/415?image=433',
  //                 dimensions: { width: 737, height: 415 }
  //               },
  //               desktop: {
  //                 alt: 'architecture',
  //                 url: 'https://picsum.photos/238/380?image=433',
  //                 dimensions: { width: 238, height: 380 }
  //               }
  //             },
  //             title: 'Hôpital',
  //             text:
  //               'Préparez au mieux votre intervention ou celle d’un proche avec notre service Hospiway. ',
  //             links: [
  //               {
  //                 url: 'https://fr.wikipedia.org/wiki/Pyramides_de_Gizeh',
  //                 label: 'En savoir plus'
  //               },
  //               {
  //                 url: 'https://fr.wikipedia.org/wiki/Pyramides_de_Gizeh',
  //                 label: 'Estimer mon reste à charge'
  //               }
  //             ]
  //           }
  //         ]}
  //       />
  //     )
  //     .toJSON();

  //   expect(tree).toMatchSnapshot();
  // });
});
