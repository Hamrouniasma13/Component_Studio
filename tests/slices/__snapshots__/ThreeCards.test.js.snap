// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <ThreeCards /> render snapshot 1`] = `
.emotion-20 {
  padding: 30px 0;
  background-color: #fff;
}

.emotion-19 {
  margin: 0px auto;
  max-width: 100%;
  padding-left: 15px;
  padding-right: 15px;
}

@media only screen and (min-width :768px) {
  .emotion-19 {
    width: 720px;
  }
}

@media only screen and (min-width :992px) {
  .emotion-19 {
    width: 960px;
  }
}

@media only screen and (min-width :1200px) {
  .emotion-19 {
    width: 1140px;
  }
}

.emotion-2 {
  margin-bottom: 30px;
}

.emotion-0 {
  display: block;
  font-family: Source Sans Pro,Arial,Helvetica,sans-serif;
  margin-bottom: 0;
  font-size: 24px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1.25em;
  font-weight: 600;
  margin-bottom: 30px;
}

@media only screen and (min-width :768px) {
  .emotion-0 {
    font-size: 28px;
    line-height: 1em;
  }
}

.emotion-18 {
  margin-left: -15px;
  margin-right: -15px;
  padding-bottom: 30px;
}

@media only screen and (min-width :768px) {
  .emotion-18 {
    margin-left: 0;
    margin-right: 0;
  }
}

.emotion-6 {
  background-color: smallLightGray;
  border: 1px solid #e5e5e5;
  height: 600px;
  overflow: hidden;
  width: 320px;
  margin: 0 auto;
  padding: 20px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #333333;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media only screen and (min-width :768px) {
  .emotion-6 {
    width: 333px;
    margin-left: 10px;
  }
}

@media only screen and (min-width :992px) {
  .emotion-6 {
    width: 350px;
    margin-left: auto;
  }
}

.emotion-1 {
  will-change: opacity;
  -webkit-transition: opacity 0.5s;
  transition: opacity 0.5s;
  opacity: 0;
  max-height: 200px;
  min-height: 140px;
  margin: 20px auto;
}

.emotion-3 {
  display: block;
  font-family: Source Sans Pro,Arial,Helvetica,sans-serif;
  margin-bottom: 0;
  font-size: 18px;
  line-height: 1.5em;
  font-weight: 600;
  font-size: 28px;
  font-weight: bold;
  margin-bottom: 30px;
}

.emotion-4 {
  text-align: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 18px;
  line-height: 1.37;
}

.emotion-5 {
  cursor: pointer;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-weight: bold;
  background-color: #494df4;
  color: #ffffff;
  font-family: Source Sans Pro,Arial,Helvetica,sans-serif;
  font-size: 15px;
  border-radius: 4px;
  -webkit-letter-spacing: 1.1px;
  -moz-letter-spacing: 1.1px;
  -ms-letter-spacing: 1.1px;
  letter-spacing: 1.1px;
  padding: 0 20px;
  position: relative;
  z-index: 0;
  overflow: hidden;
  line-height: 40px;
  text-align: center;
  max-width: 700px;
  vertical-align: middle;
  display: inline-block;
  border: none;
}

.emotion-5:after {
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  background-color: #000014;
  content: "";
  height: 530px;
  left: 50%;
  opacity: 0;
  position: absolute;
  top: 50%;
  -webkit-transform: translate3d(-50%,-50%,0) rotate(45deg);
  -ms-transform: translate3d(-50%,-50%,0) rotate(45deg);
  transform: translate3d(-50%,-50%,0) rotate(45deg);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  width: 0;
  z-index: -1;
}

.emotion-5:hover:after {
  opacity: 1;
  -webkit-transform: translate3d(-50%,-50%,0) rotate(45deg);
  -ms-transform: translate3d(-50%,-50%,0) rotate(45deg);
  transform: translate3d(-50%,-50%,0) rotate(45deg);
  width: 100%;
}

<div
  className="emotion-20"
>
  <div
    className="emotion-19"
  >
    <h1
      className="emotion-0"
      size="typo3"
    >
      My Cards
    </h1>
    <div
      className="emotion-18"
    >
      <div
        className="slick-slider slick-initialized"
      >
        
        <div
          className="slick-list"
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                "WebkitTransition": "",
                "msTransform": "translateX(0px)",
                "opacity": 1,
                "transform": "translate3d(0px, 0px, 0px)",
                "transition": "",
              }
            }
          >
            <div
              aria-hidden={false}
              className="slick-slide slick-active slick-current"
              data-index={0}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="emotion-6"
                >
                  <div
                    className="emotion-2"
                  >
                    <img
                      alt="My Easy Sante"
                      className="emotion-1"
                      src={undefined}
                    />
                  </div>
                  <h1
                    className="emotion-3"
                    size="typo5"
                  >
                    My Easy Sante
                  </h1>
                  <div
                    className="emotion-4"
                  >
                    Pour préserver et améliorer sa santé au quotidien avec My Easy Santé. C’est utile et efficace !
                  </div>
                  <div>
                    <a
                      className="Button  emotion-5"
                      color="primary"
                      href="https://fr.wikipedia.org/wiki/Pyramides_de_Gizeh"
                      onClick={[Function]}
                      size="normal"
                      target="_blank"
                    >
                      Découvrir
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active"
              data-index={1}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="emotion-6"
                >
                  <div
                    className="emotion-2"
                  >
                    <img
                      alt="Appeler un médecin 24/7"
                      className="emotion-1"
                      src={undefined}
                    />
                  </div>
                  <h1
                    className="emotion-3"
                    size="typo5"
                  >
                    Appeler un médecin 24/7
                  </h1>
                  <div
                    className="emotion-4"
                  >
                    3633 pour consulter immédiatement un médecin par téléphone (prix d'un appel local) ou par vidéo (+33 1 55 92 27 54 depuis l'étranger).
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active"
              data-index={2}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="emotion-6"
                >
                  <div
                    className="emotion-2"
                  >
                    <img
                      alt="Accéder au réseau de 6000 professionnels Santé Itelis"
                      className="emotion-1"
                      src={undefined}
                    />
                  </div>
                  <h1
                    className="emotion-3"
                    size="typo5"
                  >
                    Accéder au réseau de 6000 professionnels Santé Itelis
                  </h1>
                  <div
                    className="emotion-4"
                  >
                    6 000 professionnels de la santé Itelis conseillent et facilitent l’accès aux soins à plus de 10 millions de personnes.(opticiens, dentistes, audioprothésistes...)
                  </div>
                  <div>
                    <a
                      className="Button  emotion-5"
                      color="primary"
                      href="http://www.chateau-amboise.com/"
                      onClick={[Function]}
                      size="normal"
                      target="_blank"
                    >
                      Découvrir
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        
        
      </div>
    </div>
  </div>
</div>
`;
